pkg/manifests/manifests_test.go

```diff
@@ -46,8 +47,34 @@ func TestManifests(t *testing.T) {
 		t.Errorf("invalid DNSDaemonSet: %v", err)
 	} else {
 		// Validate the daemonset
-		if e, a := config.CoreDNSImage, ds.Spec.Template.Spec.Containers[0].Image; e != a {
-			t.Errorf("expected daemonset image %q, got %q", e, a)
+		for _, c := range ds.Spec.Template.Spec.Containers {
+			switch c.Name {
+			case "dns":
+				if e, a := config.CoreDNSImage, c.Image; e != a {
+					t.Errorf("expected daemonset dns image %q, got %q", e, a)
+				}
+			case "dns-node-resolver":
+				if e, a := config.OpenshiftCLIImage, c.Image; e != a {
+					t.Errorf("expected daemonset dns node resolver image %q, got %q", e, a)
+				}
+
+				envs := map[string]string{}
+				for _, e := range c.Env {
+					envs[e.Name] = e.Value
+				}
+				nameserver, ok := envs["NAMESERVER"]
+				if !ok {
+					t.Errorf("NAMESERVER env for dns node resolver image not found")
+				} else if *dns.Spec.ClusterIP != nameserver {
+					t.Errorf("expected NAMESERVER env for dns node resolver image %q, got %q", *dns.Spec.ClusterIP, nameserver)
+				}
+				clusterDomain, ok := envs["CLUSTER_DOMAIN"]
+				if !ok {
+					t.Errorf("CLUSTER_DOMAIN env for dns node resolver image not found")
+				} else if *dns.Spec.ClusterDomain != clusterDomain {
+					t.Errorf("expected CLUSTER_DOMAIN env for dns node resolver image %q, got %q", *dns.Spec.ClusterDomain, clusterDomain)
+				}
+			}
```

Comment:
Maybe also check that both containers, and no additional containers, are present in the daemonset.